arangod/RocksDBEngine/RocksDBSettingsManager.cpp

```diff
@@ -361,18 +366,29 @@ Result RocksDBSettingsManager::sync(bool force) {
   if (!lockForSync(force)) {
     return Result();
   }
-  TRI_DEFER(_syncing = false);
+  
+  // only one thread can enter here at a time
+
+  // make sure we give up our lock when we exit this function
+  auto guard = scopeGuard([this]() { _syncing = false; });
 
   std::unordered_map<uint64_t, CMValue> copy;
   {  // block all updates
     WRITE_LOCKER(guard, _rwLock);
     copy = _counters;
   }
 
-  rocksdb::WriteOptions writeOptions;
   // fetch the seq number prior to any writes; this guarantees that we save
   // any subsequent updates in the WAL to replay if we crash in the middle
   auto seqNumber = _db->GetLatestSequenceNumber();
+ 
+  if (seqNumber <= _lastSeqNumAfterSync) {
+    // if the RocksDB sequence number hasn't increased since we were here last,
+    // there is no need to do anything!
+    return Result();
+  }
```

Comment:
How about making `_lastSeqNumAfterSync` atomic and performing the check before acquiring the sync lock? That way you also avoid creating an unnecessary copy of `_counters`.

Reply:
Yes, sounds good.
The branch is currently some sort of proof-of-concept, and right now it does not even pass all tests. I will try to get it stable first so tests pass, and then I will care about optimizing things further.
Thanks for the suggestion!